File: 3D Touch/CommitViewController.m
//
//  CommitViewController.m
//  3D Touch
//
//  Created by Jay Versluis on 25/09/2015.
//  Copyright © 2015 Pinkstone Pictures LLC. All rights reserved.
//

#import "CommitViewController.h"

@interface CommitViewController ()

@property (nonatomic, strong) UITapGestureRecognizer *tap;

@end

@implementation CommitViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // add tap gesture recogniser
    [self.view addGestureRecognizer:self.tap];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)dismissMe {
    
    // dismisses the current view controller
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (UITapGestureRecognizer *)tap {
    
    if (!_tap) {
        _tap = [[UITapGestureRecognizer alloc]initWithTarget:self action:@selector(dismissMe)];
    }
    return _tap;
}

@end

File: 3D Touch - Swift/3D Touch Swift/CommitViewController.swift
//
//  CommitViewController.swift
//  3D Touch Swift
//
//  Created by Jay Versluis on 04/12/2015.
//  Copyright © 2015 Pinkstone Pictures LLC. All rights reserved.
//

import UIKit

class CommitViewController: UIViewController {
    
    lazy var tap: UITapGestureRecognizer = {
        let tap = UITapGestureRecognizer.init(target: self, action: "dismissMe")
        // self.view.addGestureRecognizer(tap)
        return tap
}()

    override func viewDidLoad() {
        super.viewDidLoad()

        // add tap gesture to dismiss this view
        self.view.addGestureRecognizer(tap)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func dismissMe() {
        
        // dismiss this view controller
        self.dismissViewControllerAnimated(true, completion: nil)
    }
    
}
